src/compile_error.rs

```diff
@@ -10,6 +10,9 @@ pub enum CompileError {
     #[error("Unknown type '{type_name}' at {location:?}")]
     UnknownType{type_name: String, location: core::ops::Range<usize>},
 
+    #[error("Invalid size for type : '{type_name}' at {size}")]
```

Comment:
at {size}?
do you expect to see this error ever? if this is just "theoretical" I wouldnt add a new error-type but just go with the generic code-gen error and construct the message there

Reply:
Removed the error for now, it is indeed only theoratical, we need to handle the default branch of the match